main.py

```diff
@@ -30,46 +30,93 @@ def hello():
     return 'Hello, Circles!!'
 
 
-@app.route('/circles/api/v1.0/users/', defaults={'person_id': None})
-@app.route('/circles/api/v1.0/users/<int:person_id>', methods=['GET'])
-def get_person(person_id):
+@app.route('/circles/api/v1.0/users/<int:person_id>/', defaults={'resource': None})
+@app.route('/circles/api/v1.0/users/<int:person_id>/<resource>', methods=['GET'])
+def get_person(person_id, resource):
     # Request all users.
     if not person_id:
         return jsonify(Person=get_all_nodes(Person))
-
-    # Request specific user.
+    # Fetch the person
     person = Person.match(graph, person_id).first()
     if not person:
         abort(404, description="Resource not found")
-    return jsonify(Person=person.json_repr())
-
-
-@app.route('/circles/api/v1.0/circles/', defaults={'circle_id': None})
-@app.route('/circles/api/v1.0/circles/<int:circle_id>', methods=['GET'])
-def get_circle(circle_id):
+    if not resource:
+        # Request specific user.
+        return jsonify(person.json_repr())
+
+    # Request specific resource associated with the person
+    output = []
+    if resource not in ['circles', 'events', 'knows']:
+        abort(404, description="Invalid resource specified")
+
+    if resource == 'circles':
+        for c in person.IsMember:
+            output.append(c.json_repr())
+    if resource == 'events':
+        for e in person.InvitedTo:
+            output.append(e.json_repr())
+    if resource == 'knows':
+        for k in person.Knows:
+            output.append(k.json_repr())
+    return jsonify(output)
```

Comment:
nice use of generic output!

Reply:
Ooof okay. Sorry. Upon second look, I wouldn't use an `output` variable here. If you know what resource they're requesting you know it can't be any other resource. So in each clause, just `return [k.json_repr() for k in person.Knows]`, `return [p.json_repr() for p in event.InvitedTo`]`, etc.